modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java

```diff
@@ -199,7 +199,7 @@ public void addOperationToGroup(String tag, String resourcePath, Operation opera
     public void processOpts() {
         if ((WEBCLIENT.equals(getLibrary()) && "threetenbp".equals(dateLibrary)) || NATIVE.equals(getLibrary())) {
             dateLibrary = "java8";
-        } else if (MICROPROFILE.equals(getLibrary())) {
+        } else if (MICROPROFILE.equals(getLibrary()) && "threetenbp".equals(dateLibrary)) {
```

Comment:
Is it correct to say that MICROPROFILE only supports `legacy` datetime library?

If yes, what about throwing a warning using LOGGER.warn to inform the user on that.

Reply:
The reported issue is setting dateLibrary=java8 or dateLibrary=native microprofile rest client is generating class files with Joda time because at the line 424 it has been hard coded to org.joda.time.LocalDate. So I removed line 424. Next coming to line number 202 if the user doesn't provide any dateLibrary we need to default it to native which is java.util.Date for microprofile rest client. If the user doesn't select dateLibrary open api default to threetenbp. So we are making a check at line 202 any forcing dateLibrary to native if it is microprofile & threetenbp. If user selects dateLibrary='java8' or 'joda' or 'native' this condition at line 202 will not pass and what ever user defined dateLibrary is used.

Reply:
Tested with these and the results are good.